app/commands/track/update_build_status.rb

```diff
@@ -0,0 +1,359 @@
+      submissions:,
+      mentor_discussions:,
+      syllabus:,
+      practice_exercises:,
+      test_runner:,
+      representer:,
+      analyzer:
+    }
+  end
+
+  def health
+    num_exemplar = component_health_statuses[:exemplar].to_i
+    num_healthy = component_health_statuses[:healthy].to_i
+
+    return :exemplar if num_exemplar == 5
+    return :healthy if num_exemplar + num_healthy == 5
+
+    :needs_attention
+  end
+
+  memoize
+  def component_health_statuses = [
+    analyzer_health,
+    test_runner_health,
+    representer_health,
+    syllabus_health,
+    practice_exercises_health
+  ].tally
+
+  def volunteers
+    volunteer_user_ids = User::ReputationPeriod.where(
+      period: :forever,
+      about: :track,
+      track_id: track.id
+    ).select(:user_id)
+    volunteers = User.with_attached_avatar.where(id: volunteer_user_ids).order(reputation: :desc).take(12)
+
+    {
+      num_volunteers: volunteer_user_ids.distinct.count,
+      users: SerializeAuthorOrContributors.(volunteers)
+    }
+  end
+
+  def students
+    {
+      num_students: track.num_students,
+      num_students_per_day: average_number_per_day(track.user_tracks, UserTrack)
+    }
+  end
+
+  def submissions
+    {
+      num_submissions: track.submissions.count,
+      num_submissions_per_day: average_number_per_day(track.submissions, Submission)
+    }
+  end
+
+  def mentor_discussions
+    {
+      num_discussions: track.mentor_discussions.count
+    }
+  end
+
+  def test_runner
+    status_counts = track.submissions.where(tests_status: %i[passed failed errored exceptioned]).group(:tests_status).count
+    num_passed = status_counts['passed'].to_i
+    num_failed = status_counts['failed'].to_i
+    num_errored = status_counts['errored'].to_i + status_counts['exceptioned'].to_i
+    num_test_runs = num_passed + num_failed + num_errored
+
+    {
+      num_test_runs:,
+      num_passed:,
+      num_failed:,
+      num_errored:,
+      num_passed_percentage: percentage(num_passed, num_test_runs),
+      num_failed_percentage: percentage(num_failed, num_test_runs),
+      num_errored_percentage: percentage(num_errored, num_test_runs),
+      volunteers: serialize_tooling_volunteers(track.test_runner_repo_url),
+      health: test_runner_health,
+      version: test_runner_version,
+      version_target: test_runner_version_target
+    }
+  end
+
+  memoize
+  def test_runner_health
+    # TODO: use error status to determine health (unhealthy if everything fails)
+    return :missing unless track.has_test_runner?
+    return :needs_attention if test_runner_version < 2 && track.course?
+    return :healthy if test_runner_version < 3 && track.course?
+
+    :exemplar
+  end
+
+  memoize
+  def test_runner_version = [1, Submission::TestRun.pluck(:version).last.to_i].max
+
+  def test_runner_version_target
+    return 1 unless track.has_test_runner?
+    return 2 if test_runner_version < 2
+    return 3 if test_runner_version < 3 && track.course?
```

Comment:
There is no need to upgrade to version 3 if the track does not have learning mode enabled

Reply:
Add this as a comment?